### PR TITLE
support FULL graph mode for GQA

### PR DESCRIPTION
**Pull request by momo609**
### What this PR does / why we need it?
The current library only supports the FullDecodeOnly graph mode, which enables full graph execution during the decode. This PR extends support to allow full graph execution in both the prefill and decode, referred to as FULL graph mode.
support FULL graph mode：
image

### How was this patch tested?

- vLLM version: v0.11.0
- vLLM main: https://github.com/vllm-project/vllm/commit/2918c1b49c88c29783c86f78d2c4221cb9622379

### CODE REVIEW

File: vllm_ascend/worker/model_runner_v1.py [hunk 1]
```diff
@@ -2671,12 +2676,17 @@ def _build_dummy_attn_metadata(
...
                 num_scheduled_tokens)
-            query_start_loc_tensor = torch.Tensor(cu_num_tokens).to(
-                self.device).to(torch.int32)
-            self.query_start_loc[1:num_reqs + 1] = query_start_loc_tensor
+
+            self.query_start_loc[1:num_reqs + 1] = torch.Tensor(cu_num_tokens)
             self.query_start_loc_cpu[1:num_reqs +
                                      1] = torch.Tensor(cu_num_tokens)
 
+            assigned_mask_dim = 2048
```
Comment (ApsarasX) [inline vllm_ascend/worker/model_runner_v1.py:2853]: Can we standardize all the mask size in the repository? It's confusing to have hard-coded values like 2048 used everywhere.

File: vllm_ascend/attention/mla_v1.py [hunk 2]
```diff
@@ -1461,8 +1461,7 @@ def _forward_decode(
             if workspace is None:
                 workspace = torch_npu._npu_fused_infer_attention_score_get_max_workspace(
                     q_nope, k_nope, k_nope, **common_kwargs)
-                update_graph_params_workspaces(num_tokens,
-                                               weak_ref_tensors(workspace))
+                update_graph_params_workspaces(num_tokens, workspace)
```
Comment (yiz-liu) [inline vllm_ascend/attention/mla_v1.py]: What's this for?

File: vllm_ascend/compilation/acl_graph.py [hunk 3]
```diff
@@ -446,10 +429,10 @@ def set_graph_params(aclgraph_capture_sizes: set[int]):
     )
 
 
-def update_graph_params_workspaces(num_tokens: int, workspace: Any):
+def update_graph_params_workspaces(num_tokens: int, workspace: int):
     global _graph_params
     if _graph_params is not None:
-        _graph_params.workspaces[num_tokens] = workspace
+        _graph_params.workspaces[num_tokens] = weak_ref_tensors(workspace)
```
Comment (yiz-liu) [inline vllm_ascend/compilation/acl_graph.py:435]: `weak_ref_tensors` has been refactored here, well unified.

File: vllm_ascend/attention/attention_v1.py [hunk 4]
```diff
@@ -550,6 +556,113 @@ def __init__(
...
+                input_layout="TND",
+                block_size=block_size,
+                actual_seq_lengths=query_start_loc,
+                actual_seq_lengths_kv=actual_seq_lengths_kv,
+                num_key_value_heads=self.num_kv_heads,
+                num_heads=self.num_heads,
+                sparse_mode=3,
+                scale=self.scale,
+            )
+            update_graph_params_workspaces(num_tokens, workspace)
```
Comment (yiz-liu) [inline vllm_ascend/attention/attention_v1.py:624]: We should make `workspace` a more general param, after all, it can be used in eager mode as well.

File: vllm_ascend/attention/attention_v1.py [hunk 5]
```diff
@@ -550,6 +556,113 @@ def __init__(
...
+            scale=self.scale,
+            sparse_mode=3,
+            workspace=workspace,
+            out=[output, softmax_lse],
+        )
+
+        output = output.view(num_tokens, self.num_heads, self.head_size)
+
+        handle = torch.npu.graph_task_group_end(stream)
+        graph_params.handles[num_tokens].append(handle)
```
Comment (yiz-liu) [inline vllm_ascend/attention/attention_v1.py:663]: This can be refactored to be a `context_manager`:
```suggestion
    with init_attn_graph_task(graph_mode):
        torch_npu.npu_fused_infer_attention_score.out(
            query=query,
            key=key,
            value=value,
            atten_mask=attn_metadata.fia_attn_mask,
            block_table=block_table,
            input_layout="TND",
            block_size=block_size,
            actual_seq_lengths=query_start_loc,
            actual_seq_lengths_kv=actual_seq_lengths_kv,
            num_key_value_heads=self.num_kv_heads,
            num_heads=self.num_heads,
            scale=self.scale,
            sparse_mode=3,
            workspace=workspace,
            out=[output, softmax_lse],
        )
```
so that we can easily unify the code path of eager mode and graph mode.

File: vllm_ascend/worker/model_runner_v1.py [hunk 6]
```diff
@@ -1738,6 +1744,7 @@ def _prepare_inputs(
         self.attn_mask = self._make_attention_mask(seq_lens=seq_lens_cpu,
                                                    position=positions_cpu,
                                                    attn_state=attn_state)
+        self.fia_attn_mask = self._make_fia_attention_mask()
```
Comment (weijinqian0) [inline vllm_ascend/worker/model_runner_v1.py:1747]: why need two attn_mask?

**Review by weijinqian0 — APPROVED** [review]

File: vllm_ascend/worker/model_runner_v1.py (hunk 6 above)
Comment (momo609) [reply vllm_ascend/worker/model_runner_v1.py]: > why need two attn_mask?
fullgraph all use fia，will remove in next pr.
